hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -556,17 +556,26 @@ private Pair<Integer, Long> deleteTransactions(
             LOG.warn("Missing delete block(Container = " +
                 container.getContainerData().getContainerID() + ", Block = " +
                 blkLong);
+            blocksDeleted++;
             continue;
           }
+
           try {
             handler.deleteBlock(container, blkInfo);
             blocksDeleted++;
-            bytesReleased += KeyValueContainerUtil.getBlockLength(blkInfo);
-          } catch (InvalidProtocolBufferException e) {
-            LOG.error("Failed to parse block info for block {}", blkLong, e);
           } catch (IOException e) {
+            // TODO: if deletion of certain block retries exceed the certain
+            //  number of times, service should skip deleting it,
+            //  otherwise invalid numPendingDeletionBlocks could accumulate
+            //  beyond the limit and the following deletion will stop.
```

Comment:
Have we run into an issue where block was throwing an error on deletion by the handler?

Reply:
> Have we run into an issue where block was throwing an error on deletion by the handler?

Right now we have not met such a case, but it's a potential risk since the exception handle is not well. Anything affecting the `blocksDeleted` here could have an Avalanche effect.